src/Symfony/Component/Serializer/Tests/Encoder/CsvEncoderTest.php

```diff
@@ -135,12 +135,42 @@ public function testEncodeEmptyArray()
         $this->assertEquals("\n\n", $this->encoder->encode(array(array()), 'csv'));
     }
 
-    /**
-     * @expectedException \Symfony\Component\Serializer\Exception\InvalidArgumentException
-     */
-    public function testEncodeNonFlattenableStructure()
+    public function testEncodeVariableStructure()
+    {
+        $value = array(
```

Comment:
What about:

```php
array(
'a' => array('a', 'b'),
'a' => 'foo',
)
```

This case looks not easy to handle, it's one of the reasons of the current behavior.

Reply:
actually this one gets handled just fine, but the result may be a little bit unexpected:
```
a.0,a.1,a
a,b,
,,foo

```

i dont think we need to handle this differently in the encoder, because that one is indeed a custom normalizers job

Reply:
The output is a bit weird but why not.

Reply:
btw this is a more a problem of the flattening algorithm, which is untouched by the PR.
compare it with the previously support case of:
```php
[
  'a' => ['foo', 'bar'],
  'a.0' => 'baz',
]
```